producer.go

```diff
@@ -151,8 +151,10 @@ const (
 
 // MessageToSend is the collection of elements passed to the Producer in order to send a message.
 type MessageToSend struct {
-	Topic      string
-	Key, Value Encoder
+	Topic    string      // The Kafka topic for this message
+	Key      Encoder     // The partitioning key for this message. It must implement the Encoder interface. Pre-existing Encoders include StringEncoder and ByteEncoder
+	Value    Encoder     // The actual message to store in Kafka. It must implement the Encoder interface. Pre-existing Encoders include StringEncoder and ByteEncoder
+	Metadata interface{} // This field is used to hold arbitrary data you wish to include so it will be available when recieving on the Succcesses and Errors channels
```

Comment:
typo in "Succcesses" - also it would probably be good to mention that Sarama completely ignores this field so it's clear it's just for pass-through data